dfsclient/src/main/java/org/hpccsystems/dfs/client/HpccRemoteFileReader.java

```diff
@@ -75,6 +75,24 @@ public static class FileReadContext
         public int readBufferSizeKB = -1;
         public int readRequestSpanBatchSize = -1; // The number of read requests before creating a new span
         public Span parentSpan = null;
+
+        private long getReadSizeKB()
+        {
+            if (readSizeKB <= 0)
```

Comment:
should we ever allow readSizeKB to be <=0 ?

Reply:
In this case we are treating -1 as a "use the default value" indicator so we need to check for < 0 and default